cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -489,17 +489,13 @@ TEST(GcsFileSystem, FileSystemCompare) {
   a_options.project_id = "test-only-invalid-project-id";
   auto a = GcsFileSystem::Make(a_options);
   EXPECT_THAT(a, NotNull());
-  EXPECT_TRUE(a->Equals(*a));
```

Comment:
Why do we need to remove this?

Should we change `auto a = GcsFileSystem::Make(a_options);` to `ASSERT_OK_AND_ASSIGN(auto a, GcsFileSystem::Make(a_options);` instead?

Reply:
There is no need to remove this with your suggested change. That change has been made, and the above line has been restored. Ref 4d6687ad52ac1734fab38e092ebc149fe87b09ce.